File: lua/muninn/tests/fixtures/lang.ts
// basic function
function greet(): void {
	console.log("hello");
}

// typed parameters
function add(a: number, b: number): number {
	return a + b;
}

// interface definition
interface Point {
	x: number;
	y: number;
}

// type alias
type Direction = "north" | "south" | "east" | "west";

// class with types
class Vector {
	constructor(public x: number, public y: number) {}

	magnitude(): number {
		return Math.sqrt(this.x ** 2 + this.y ** 2);
	}
}

// const arrow function
const multiply = (a: number, b: number): number => {
	return a * b;
};

// enum declaration
enum Color {
	Red,
	Green,
	Blue,
}

// generic function
function identity<T>(arg: T): T {
	return arg;
}

// async function
async function fetchData(url: string): Promise<Response> {
	return fetch(url);
}

// global variable
var globalCount: number = 42;

File: lua/muninn/tests/fixtures/lang.c
#include <stdio.h>

// basic function
void greet(void) {
	printf("hello\n");
}

/*
 * block comment
 */
int add(int a, int b) {
	return a + b;
}

// struct definition
struct point {
	int x;
	int y;
};

// enum definition
enum direction {
	NORTH,
	SOUTH,
	EAST,
	WEST
};

// union definition
union value {
	int i;
	float f;
};

// typedef
typedef struct {
	int x;
	int y;
} vec2;

// function pointer typedef
typedef void (*callback)(int);

// static function
static void helper(void) {
	printf("static\n");
}

// forward declaration
void forward(void);

// global variable
int global_count = 42;
